### PR TITLE
Make cropping data available for rawspeed sraws

### LINKED ISSUE #15064: DNG Files from the Sigma SD Quattro are not rendered correctly
### Describe the bug

In DNG files from the Sigma SD Quattro there are black and purple borders around the left and upper edge of the picture:

image

I'm making the zipped original DNG available here:
https://www.dropbox.com/scl/fi/q1qbad9qgiu96ofcic798/SDQ_0453.zip?rlkey=7x39cq7vc2vmmv8lh746cvdyg&dl=0

If needed, I can supply more test images.

### Steps to reproduce

1. Download SDQ_0453.zip and import the resulting SDQ_0453.DNG image
2. Open the image in the "darkroom" view

### Expected behavior

Darktable should render the image without black borders and purple and green artifacts.

### Logfile | Screenshot | Screencast

_No response_

### Commit

_No response_

### Where did you install darktable from?

darktable.org

### darktable version

4.4.1

### What OS are you using?

Linux

### What is the version of your OS?

Ubuntu 23.04

### Describe your system?

_No response_

### Are you using OpenCL GPU in darktable?

No

### If yes, what is the GPU card and driver?

_No response_

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

_No response_

### LINKED ISSUE #15069: DNG ActiveArea instructs to crop on top, but darktable crops on bottom
### Describe the bug

Hi, developer of https://github.com/dnglab/dnglab here. I'm working on X-Trans Pixel-Shift merging (Fuji X-T5 and the like) and discovered a strange problem in darktable with the resulting DNGs. It has not relation to the merged pixel data but the ActiveArea tag in general. For testing purposes, I've set the ActiveArea to some extreme values (cropping 3500 pixels from top):

```
exiv2 -pR fail.dng | grep ActiveArea
  590599330 | 0xc68d ActiveArea                   |     SHORT |        4 | 590599092 | 3500 0 10392 15744
```
The first value is the crop border from top, so the expected result should be cropped on top. But the opposite is true, the image bottom is truncated:
grafik

Full image without any cropping by ActiveArea:

grafik

So for some reason, darktable always crops the image on bottom. I've modified the ActiveArea to "really" crop 3500 pixels from the bottom instead from top to verify if the processing is just inversed. But it's not. Darktable always crops bottom.

I've checked rawspeed DNG decoder, the extracted rectangle coordinates are correct. So I assume the problem is located somewhere else but I was not able to figure it out.
The DNG image is properly cropped and displayed in Adobe Camera btw.

### Steps to reproduce

1. Download DNG file from https://chaospixel.com/pub/github/dt/activeareabug/fail.dng
2. Open DNG file in darktable

### Expected behavior

darktable should crop as ActiveArea tag advises (crop on top for the example image)

### Logfile | Screenshot | Screencast

_No response_

### Commit

_No response_

### Where did you install darktable from?

self compiled

### darktable version

4.3.0+2514~gcd43e9e070 and 4.5.0+328~g1aca07c62d

### What OS are you using?

Linux

### What is the version of your OS?

Debian

### Describe your system?

One System with OpenCL, the other without OpenCL.

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

nvidia

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

_No response_

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
Until now we only took data from the top of image as the cropping info provided by active area and default crop was not respected.

Fixes #15069
Fixes #15064

### CODE REVIEW

**Comment by kmilos** [comment]
Thanks, I'll check on https://github.com/darktable-org/darktable/issues/15064 as well.

**Comment by jenshannoschwalm** [comment]
Yes the cropping issue for sigma has the same reason.

**Comment by kmilos** [comment]
Confirmed working for the Sigma sd Quattro DNG, thanks. Also works on the Pentax K-3 III Monochrome DNG.

**Comment by LebedevRI** [comment]
Is that not what the normal code path does, why code duplication?
Can't you just move the original code block higher up before the call to this function?

**Comment by jenshannoschwalm** [comment]
Fixed according to suggests.

**Review by kmilos — APPROVED** [review]
LGTM

File: src/imageio/imageio_rawspeed.cc [hunk 1]
```diff
@@ -529,6 +528,7 @@ dt_imageio_retval_t dt_imageio_open_rawspeed_sraw(dt_image_t *img, RawImage r, d
           for(int k = 0; k < 3; k++)
           {
             out[k] = (float)in(j, cpp * i + k) / (float)UINT16_MAX;
+            out[3] = 0.0f;
```
Comment (da-phil) [inline src/imageio/imageio_rawspeed.cc:531]: I do not get why you put loop-independent constant assignment into a nested for-loop, also in line 551. Can't you just move those lines below the lines where ``out`` is defined?

Comment (jenshannoschwalm) [reply src/imageio/imageio_rawspeed.cc]: just stupid me. Thanks!

Comment (da-phil) [reply src/imageio/imageio_rawspeed.cc]: ah, okay, I thought I was stupid too, not getting the big picture :D

File: src/imageio/imageio_rawspeed.cc [hunk 2]
```diff
@@ -527,9 +526,8 @@ dt_imageio_retval_t dt_imageio_open_rawspeed_sraw(dt_image_t *img, RawImage r, d
         for(int i = 0; i < img->width; i++, out += 4)
         {
           for(int k = 0; k < 3; k++)
-          {
             out[k] = (float)in(j, cpp * i + k) / (float)UINT16_MAX;
-          }
+          out[3] = 0.0f;
```
Comment (da-phil) [inline src/imageio/imageio_rawspeed.cc:530]: I see that you now you put it into the outer loop, not outside just below the definition of ``out``. Do you expect the address of ``out`` to be changed from the outside?

Comment (jenshannoschwalm) [reply src/imageio/imageio_rawspeed.cc:530]: We iterate through width and for each position we write the three colors into the 4-float-per-pixel data used later in the pipeline. The `out[3] = 0.0f` ensures the 'mask channel' to be _defined_ and never be a NaN for every pixel.

Comment (da-phil) [reply src/imageio/imageio_rawspeed.cc:530]: okaaaay, now I understand, nevermind my dumb comment then :see_no_evil:

**Review by TurboGit — APPROVED** [review]
Thanks !

Need a release notes entry please. TIA.

**Comment by jenshannoschwalm** [comment]
@TurboGit Release note: Fixed wrong cropping of sensor data for sraw dng files

**Pull request merged**